pkg/operator/ceph/csi/ceph_config.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	configName = "csi-ceph-conf-override"
+
+	cephCongKey = "ceph.conf"
+	cephConfVal = `[global]
+auth_cluster_required = cephx
+auth_service_required = cephx
+auth_client_required = cephx
+
+# Workaround for http://tracker.ceph.com/issues/23446
+fuse_set_user_groups = false
+
+# ceph-fuse which uses libfuse2 by default has write buffer size of 2KiB
+# adding 'fuse_big_writes = true' option by default to override this limit
+# see https://github.com/ceph/ceph-csi/issues/1928
+fuse_big_writes = true
+`
+	// keyring should be adds to the same configmap as ceph.conf as we cannot
+	// create keyring file in the same directory as ceph.conf.
+	keyringKey = "keyring"
+	// empty keyring file content
+	keyringVal = ""
+)
+
+// CreateCsiConfigMap creates an ceph.conf for ceph-csi. If a config map already
+// exists, it will return it.
```

Comment:
So if someone wants a custom ceph.conf in the csi driver they have to create it first, or update the default after it is created here?

Reply:
What if the default needs to change in the future? It will be difficult to know the difference between a previous default and user-customized contents. You might consider for the csi driver to merge its required settings with the user-specified settings, and keep this configmap empty by default for user settings.

Reply:
>So if someone wants a custom ceph.conf in the csi driver they have to create it first, or update the default after it is created here?

both works if the configmap is in the expected format.

>What if the default needs to change in the future? It will be difficult to know the difference between a previous default and user-customized contents. You might consider for the csi driver to merge its required settings with the user-specified settings, and keep this configmap empty by default for user settings.

merging might be difficult one but we can document the new configuration that needs to be added as part of the upgrade process?

Reply:
I don't think we should mix user settings and required settings in the same configmap or else we really can't control what happens during upgrade. We have run into this with the OCS operator and the existing ceph.conf overrides. If they want ceph.conf overrides for the csi driver, we need to reserve this config purely for the user config. Are the ceph client defaults not sufficient if we leave this blank?

Reply:
@travisn any idea about this one? it's difficult to merge I think. or can we introduce ceph.conf configmap per ceph cluster to override defaults? if user creates a configmap in the ceph cluster namespace with the default name we will get and replace the content of ceph.conf we are creating.  any better idea how to solve this one?

Reply:
What about the following?
- Create an empty configmap by default? 
- If the CSI driver finds an empty configmap, set to its default
- If the CSI driver finds settings in the configmap, use it (and don't set the default)

This way the default stays with the csi driver, and the user can override to anything and we don't need to worry about merging. But the main advantage I see is that the default can be updated. This PR doesn't allow the default to ever be updated since we won't reconcile the configmap after creation.

Reply:
due to mounting configmap as a file cephcsi cannot write back to the configmap? let me do double check if its possible

Reply:
The CSI driver would need to mount this configmap to some location other than `/etc/ceph/ceph.conf` so that the csi driver could either copy from the other location to here, or the csi driver would write its default just like it was doing previously.